bin/porta

```diff
@@ -493,11 +494,11 @@ module Porta
     end
 
     def run_porxy
-      system("docker run -d --name porxy --rm -p 3008:3008 #{options[:porxy_image]} >/dev/null && echo \"porxy\"")
+      system("docker run -d --name porxy --rm --add-host=host.docker.internal:host-gateway -p 3008:3008 #{options[:porxy_image]} >/dev/null && echo \"porxy\"")
```

Comment:
I'm afraid this will break in macos:

```172.17.0.1 - apicastsecret [20/Apr/2021:09:11:27 +0000] "GET /master/api/proxy/configs/staging.json?host=api-2.staging.apicast.dev HTTP/1.1" 502 158 "-" "APIcast/3.6.0 (Linux; x64; env:staging) APIcast/builtin" "-"
2021/04/20 09:11:58 [error] 14#14: *1 connect() failed (110: Connection timed out) while connecting to upstream, client: 172.17.0.1, server: _, request: "GET /master/api/proxy/configs/admin/api/services.json HTTP/1.1", upstream: "http://192.168.65.1:3000/master/api/proxy/configs/admin/api/services.json", host: "host.docker.internal:3008"
172.17.0.1 - apicastsecret [20/Apr/2021:09:11:58 +0000] "GET /master/api/proxy/configs/admin/api/services.json HTTP/1.1" 502 158 "-" "APIcast/3.6.0 (Linux; x64; env:staging) APIcast/builtin" "-"
172.17.0.1 - apicastsecret [20/Apr/2021:09:12:08 +0000] "GET /master/api/proxy/configs HTTP/1.1" 499 0 "-" "APIcast/3.6.0 (Linux; x64; env:staging) APIcast/builtin" "-"
2021/04/20 09:15:15 [error] 14#14: *7 connect() failed (110: Connection timed out) while connecting to upstream, client: 172.17.0.1, server: _, request: "GET /master/api/proxy/configs/staging.json?host=api-2.staging.apicast.dev HTTP/1.1", upstream: "http://192.168.65.1:3000/master/api/proxy/configs/staging.json?host=api-2.staging.apicast.dev", host: "host.docker.internal:3008"
```

Maybe this would do:
```suggestion
      system("docker run -d --name porxy --rm #{'--add-host=host.docker.internal:host-gateway' if linux?} -p 3008:3008 #{options[:porxy_image]} >/dev/null && echo \"porxy\"")
```

combined of course with:

```diff
diff --git a/bin/porta b/bin/porta
index f5ac2eb..7be88eb 100755
--- a/bin/porta
+++ b/bin/porta
@@ -406,6 +406,14 @@ module Porta
     def stdout_log?
       options[:stdout_log]
     end
+
+    def macos?
+      /darwin/ =~ RUBY_PLATFORM
+    end
+
+    def linux?
+      !macos?
+    end
   end
 
   class ServerCommand < CommandRunner
```

Reply:
That sounds better, I will change that :)